src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -358,5 +343,163 @@ public ResponseEntity<List<LitePermissionDTO>> retrievePermissionsByUsername(@Pa
+		    LOGGER.info("No settings for the current user {}", userName);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		List<UserSettingDTO> userSettingsDTO = userSettingMapper.map2DTOList(userSettings);
+		LOGGER.info("Found {} user settings", userSettingsDTO);
+		return ResponseEntity.ok(userSettingsDTO);
+	}
+	
+	/**
+	 * Returns a {@link UserSettingDTO} of userSetting created.
+	 * 
+	 * @param userSettingDTO -  the {@link UserSettingDTO} to insert.
+	 * @return {@link UserSettingDTO} if the userSetting has been created, null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@PostMapping(value = "/users/settings", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> newUserSettings(@Valid @RequestBody UserSettingDTO userSettingDTO) throws OHServiceException {
+		LOGGER.info("Create a UserSetting");
+		String userName = userSettingDTO.getUser();
+		String user = SecurityContextHolder.getContext().getAuthentication().getName();
+		userSettingDTO.setId(0);
+		final String ADMIN = "admin";
+		if (!userName.equals(user) 
+						&& !user.equals(ADMIN)) {
+		    throw new OHAPIException(new OHExceptionMessage("Not allowed."));
+		}
+		UserSetting userSetting = userSettingMapper.map2Model(userSettingDTO);
+		UserSetting created = userSettingManager.newUserSetting(userSetting);
+		if (created == null) {
+		    LOGGER.info("UserSetting is not created!");
+		    throw new OHAPIException(new OHExceptionMessage("UserSetting not created."));
+		}
+		LOGGER.info("UserSetting successfully created!");
+		return ResponseEntity.status(HttpStatus.CREATED).body(userSettingMapper.map2DTO(created));
+	}
+	
+	/**
+	 * Updates an existing {@link UserSettingDTO}.
+	 * 
+	 * @param userSettingDTO - the {@link UserSettingDTO} to update.
+	 * @param id - id of {@link UserSetting} .
+	 * @return {@link UserSettingDTO} if the UserSetting has been updated , null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@PutMapping(value = "/users/settings/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> updateUserSettings(@PathVariable(name = "id") int id, @Valid @RequestBody UserSettingDTO userSettingDTO) throws OHServiceException {
+		LOGGER.info("Update a UserSetting");
+		String userName = userSettingDTO.getUser();
+		String user = SecurityContextHolder.getContext().getAuthentication().getName();
+		final String ADMIN = "admin";
+		if (userSettingDTO.getId() == 0 || (userSettingDTO.getId() != 0 && userSettingDTO.getId() != id)) {	
+		    throw new OHAPIException(new OHExceptionMessage("Malformed request."));
+		}
+		Optional<UserSetting> userSetting = userSettingManager.getUserSettingById(id);
+		UserSetting updated;
+		if (!userSetting.isPresent()) {
+		    LOGGER.info("No user settings with id {}", id);
+	        throw new OHAPIException(new OHExceptionMessage("UserSetting doesn't exist."));
+		}
+		if (!userSetting.get().getUser().equals(userSettingDTO.getUser()) &&
```

Comment:
leveraging local variable:

```suggestion
		if (!userSetting.get().getUser().equals(requestUserName) &&
```